topologies/binding/binding_test.go

```diff
@@ -218,3 +221,38 @@ func TestReservation_Error(t *testing.T) {
 		}
 	}
 }
+
+func TestDialOTGTimeout(t *testing.T) {
+	const timeoutSecs = 42
+	a := &staticATE{
+		AbstractATE: &binding.AbstractATE{Dims: &binding.Dims{Name: "my_ate"}},
+		r:           resolver{&bindpb.Binding{}},
+		dev: &bindpb.Device{Otg: &bindpb.Options{
+			Timeout: timeoutSecs,
+		}},
+	}
+	grpcDialContextFn = func(context.Context, string, ...grpc.DialOption) (*grpc.ClientConn, error) {
+		return nil, nil
+	}
+	gosnappiNewAPIFn = func() gosnappi.GosnappiApi {
+		return &captureAPI{GosnappiApi: gosnappi.NewApi()}
+	}
+	api, err := a.DialOTG(context.Background())
```

Comment:
Do we need a `defer` for these that restores the previous state for future unit tests?